src/NzbDrone.Common/Http/HappyEyeballs/HappyEyeballs.cs

```diff
@@ -0,0 +1,182 @@
+/*
+Until .NET implements Happy Eyeballs natively, use third-party implementation from
+https://slugcat.systems/post/24-06-16-ipv6-is-hard-happy-eyeballs-dotnet-httpclient/#the-implementation
+This issue is being tracked at https://github.com/dotnet/runtime/issues/26177.
+
+Below is a slightly modified Happy Eyeballs implementation from the post above.
+We’ve factored out HTTP-specific implementation into HttpHappyEyeballs class to
+make testing easier.
+*/
+
```

Comment:
For consistency it'd be better to move this into the namespace, still prominent, but keeps imports at the top.